inst/tutorials/chapter3/chapter3.Rmd

```diff
@@ -145,32 +173,156 @@ y <- rlnorm(10000,meanlog=3.5,sdlog=1)
+               
+               conditionalPanel(condition = "input.dist == 'rationorm'",
+                                sliderInput("a", "Shift Numerator", min = -5, max = 5, step = .1, value = -2),
+                                sliderInput("b", "Shift Denominator", min = -5, max = 5, step = .1, value = -3)),
+               
+               conditionalPanel(condition = "input.dist == 'rlnorm'",
+                                sliderInput("mean_lnorm", "Mean", min = -5, max = 5, step = .1, value = 0),
+                                sliderInput("std_lnorm", "Standard Deviation", min = 0.01, max = 5, step = 0.1, value = 1))),
+  
+  mainPanel(
+    plotOutput("dPlot") #plot
+  )))
+```
+
+```{r, context="server"}
+output$dPlot <- renderPlot({
+  
+  if (input$dist == 'rnorm'){ # Normal
+    range <- seq(-10, 10, .1)
+    plot(range,
+         dnorm(range, mean = input$mean_norm, sd = input$std_norm), "l", col = "navyblue", ylab = "Density", xlab = "y", main = "Normal Distribution")
+    abline(v=input$mean_norm ,col="red",lw=3)
+    abline(v=input$mean_norm ,col="blue",lw=1)
+    legend("topright",c("mean","median"),col=c("red","blue"),lty=c(1,1),lw=c(2,2))
+    
+  } else if(input$dist == 'rlnorm'){ # Log Normal
+    range <- seq(0, 10, .1)
+    plot(range,
+         dlnorm(range, mean = input$mean_lnorm, sd = input$std_lnorm), "l", col = "navyblue", ylab = "Density", xlab = "y", main = "Lognormal Distribution")
+        abline(v=exp(input$mean_lnorm + (input$std_lnorm^2)/2) ,col="red",lw=3)
+        abline(v=exp(input$mean_lnorm) ,col="blue",lw=1)
+        legend("topright",c("mean","median"),col=c("red","blue"),lty=c(1,1),lw=c(2,2))
+
+
+  } else if(input$dist == 'runif'){ #uniform
+    range <- seq(input$range_unif[1], input$range_unif[2], .1)
+    plot(range,
+         dunif(range, min = input$range_unif[1], max = input$range_unif[2]), "l", col = "navyblue", ylab = "Density", xlab = "y", main = "Uniform Distribution")
+    abline(v=.5*input$b - .5*input$a,col="red",lw=3)
+    abline(v=.5*input$b - .5*input$a,col="blue",lw=1)
+    legend("topright",c("mean","median"),col=c("red","blue"),lty=c(1,1),lw=c(2,2))
+    
+  } else if(input$dist == 'rbeta') { #Beta
+    range <- seq(0.001, .999, .001)
+    plot(range,
+         dbeta(range, shape1 = input$alpha, shape2 = input$beta), "l", col = "navyblue", ylab = "Density", xlab = "y", main = "Beta Distribution")
+    abline(v=(input$alpha/ (input$alpha + input$beta)), col="red",lw=3)
+    #abline(v=input$location * 2^(1/input$shape) ,col="blue",lw=1) Median of the Beta function -> no general closed-form...
+    legend("topright",c("mean","median"),col=c("red","blue"),lty=c(1,1),lw=c(2,2))
+    
+  } else if(input$dist == 'rparet'){ #Pareto
+    range <- seq(0, 10, .1)
+    plot(range,
+         dpareto(range, location = input$location, shape = input$shape), "l", col = "navyblue", ylab = "Density", xlab = "y", main = "Pareto Distribution")
```

Comment:
it would be cool to add a comment below each plot, if there is anything noteworthy to say. here for example
```R
dpareto(range, location = input$location, shape = input$shape), "l", col = "navyblue", ylab = "Density", xlab = "y", main = "Pareto Distribution",sub="Observe the zero mass left of `location`, and the very long right tail")
```

Reply:
Done, but overlap btw subtitles and axis...